### ISSUE TITLE
WI classifier (CNN) training

### ISSUE THREAD

**Issue by chunkypilu** [luizgh/sigver_wiwd#8]
Dear Sir,

Please reply these questions regarding training WI classifier (CNN), 
1.Both real and forgeries of a user form development set used? ,and if both were used then were they  given same label or different?
2.Were all forgeries for all users given same label?

Thanks,
Chunky

**Comment by luizgh** [comment]
In this paper we evaluated different formulations of the problem, considering scenarios where only genuine signatures are available, and scenarios where both genuine signatures and forgeries are used, see section 3 of the paper:

**3.1 Learning feature from genuine signatures**: only genuine signatures were used. In the paper, we refer to this as "SigNet"
**3.2.1 Treat forgeries as separate classes**: in this formulation, forgeries for different users have each a different label, so you have twice as many classes as the number of users.
**3.2.2 Add a separate output for detecting forgeries**: in this formulation, we consider two objective functions: learning to discriminate the user, and learning to separate genuine signatures and forgeries. That is, the dataset is now of the format (X, y, f), where X is the signature, y is the class (user) and f is a binary variable that indicates if the signature is a forgery. The forgeries have the same class (y) as the genuine. Note, however, that the loss function that worked best ("L2", equation (4) in the paper) does not consider the label y of the forgeries (that is, genuine signatures contribute to both losses, while forgeries only contribute to the "forgery classification loss"). In the paper, we refer to this model as "SigNet-F"

I believe this answers the first question. For the second question ("Were all forgeries for all users given same label?"): we have not tried this alternative in this paper

**Comment by chunkypilu** [comment]
ok.
